pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -75,6 +76,12 @@ func newECRProvider(getterFactory tokenGetterFactory, isEC2 ec2ValidationFunc) *
 
 // Enabled implements DockerConfigProvider.Enabled.
 func (p *ecrProvider) Enabled() bool {
+	if credentialprovider.AreLegacyCloudCredentialProvidersDisabled() {
+		warnOnce.Do(func() {
+			klog.V(4).Infof("AWS credential provider is now disabled. Please refer to sig-cloud-provider for guidance on external credential provider integration for AWS")
```

Comment:
this will surface logs on all platforms, even non-aws ones, right? are AWS credential provider logs currently emitted at v(4) on non-aws platforms? if not, I don't think we should start now.

would it make more sense to do this in the Provide method when we know we're actually on AWS (once we pass the `isEC2` check?

Reply:
My understanding is that the AWS credential provider [silently no-ops](https://github.com/kubernetes/kubernetes/blob/5932233d651dd8d7c5703cf1183c0dd1f2a583d5/pkg/credentialprovider/aws/aws_credentials.go#L161-L173) if it decides it's not running on an EC2 instance. Since the GCP credential provider already does something similar in this PR (logging the feature gate message only if`onGCE` has the right value), it seems like a good trend to follow.

Reply:
Good point. Done.